services/worker-manager/src/worker-scanner.js

```diff
@@ -63,6 +64,16 @@ class WorkerScanner {
           this.monitor.info(
             `Worker ${worker.workerGroup}/${worker.workerId} has unknown providerId ${worker.providerId} (ignoring)`);
         }
+
+        // If the worker will be expired soon but it still exists,
+        // update it to stick around a while longer. If this doesn't happen,
+        // long-lived instances become orphaned from the provider. We don't update
+        // this on every loop just to avoid the extra work when not needed
+        if (worker.expires < taskcluster.fromNow('1 day')) {
+          await worker.update(this.db, worker => {
+            worker.expires = taskcluster.fromNow('1 week');
+          });
+        }
```

Comment:
Now that I look at this logic, it's not doing what we want -- we'd like to have every active worker have an expiration at least 1 week in the future.  This logic lets expirations range from 1 to 7 days in the future.  I think we'd want them to range from 7 to 8 days in the future, instead (thus updating them once a day on average).

That should be a quick change to this logic, and updates to the tests -- do you want to just make that adjustment here, or file a followup?

Reply:
I updated that logic and the tests.